xbmc/addons/BinaryAddonCache.h

```diff
@@ -31,8 +31,9 @@ namespace ADDON {
 class CBinaryAddonCache : public Observer
 {
 public:
-  virtual ~CBinaryAddonCache();
+  virtual ~CBinaryAddonCache() { }
```

Comment:
I don't agree. The lifetime of a singleton is infinity because it is static. An object embedded into another object does not need to be aware of this.